src/dvc_data/hashfile/state.py

```diff
@@ -98,10 +98,15 @@ def save(self, path, fs, hash_info):
         if not isinstance(fs, LocalFileSystem):
             return
 
+        if fs.version_aware:
+            _, version_id = fs.path.split_version(path)
+        else:
+            version_id = None
```

Comment:
State does not support non-local filesystems, so I don't think we need to add support for this. It's a dead code. :)

Reply:
My understanding is that this is needed for some of the things @efiop is working on

Reply:
Yeah, let's keep this (using this in POCs, but could probably turn this on later too).

Reply:
I'd prefer not adding this dead code. In POC, you can cherry-pick easily, no need to add to main.
We don't even support non-local filesystems, version-aware fs is too far.

Reply:
Though version_ids are problematic for state in https://github.com/iterative/dvc-data/pull/153/files#r958293617 so we might indeed not touch it in this PR.